GoogleMapsComponents/BlazorGoogleMapsKeyService.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System.Threading.Tasks;
+
+namespace GoogleMapsComponents;
+
+/// <summary>
+/// Interface for the key service that is injected into the Map Component in order to retrieve
+/// a google api key when injecting the google api JS package
+/// </summary>
+public interface IBlazorGoogleMapsKeyService
+{
+    public Task<Maps.MapApiLoadOptions> GetApiOptions();
+
+    public bool IsApiInitialized { get; set; }

```

Comment:
Whats other use case except in MapComponent?
I think you could remove here and have private bool property in map component. Fix me if i am wrong

Reply:
The use of IsApiInitialized is to prevent attempting to initialize the API twice. The only place that information can reasonably be stored is in the scoped service because that is more or less a single instance for the lifetime of the connection in which the google API has been loaded in the browser page. If we stored it as private in MapComponent, then it wouldn't persist either to a 2nd MapComponent instance on the page or to a different page in the SPA and returns to a new MapComponent instance. 

And of course if we removed it completely then every time within a single Blazor connection that you come to a new page with a map, the call to bootstrap load the google api will happen all over again which adds an async/await over SignalR delay to performance and a browser console warning from google.

I think there may also be a use case outside of MapComponent if the user genuinely does need to try to reload the google API within the same scope for some reason (adding more google libraries on a different page?) but I haven't tested how that works.

Finally, it needs to be a `public` property of the interface/service class because user implementations of the service class need to make it accessible by MapComponent, so something like `internal` wouldn't work.